### PR TITLE
V3: Add TextInput component

### PR DESCRIPTION
**Pull request by tuanddd**
- [x] Add `TextInput` component
- [x] Add support for speech recognition
- [x] Add search icon
- [x] Add loading icon

### CODE REVIEW

**Comment by tuanddd** [comment]
Let me check again if it's can be rebased.

File: src/components/TextInput/icons.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import React from "react";
```
Comment (sampotts) [inline src/components/TextInput/icons.tsx:1]: With the svgs, how about we put them in a folder (`/src/assets/icons` or something?) as svg files and have an `index.tsx` that has a heap of exports? e.g. `export { ReactComponent as IconName } from './icon-name.svg';`?

File: src/components/TextInput/icons.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    xmlns="http://www.w3.org/2000/svg"
+    viewBox="0 0 16 16"
+    width="16"
+    height="16"
+    {...props}
+  >
+    <g className="nc-icon-wrapper">
+      <path d="M8,10c2.2,0,4-1.8,4-4V4c0-2.2-1.8-4-4-4S4,1.8,4,4v2C4,8.2,5.8,10,8,10z M6,4c0-1.1,0.9-2,2-2s2,0.9,2,2v2 c0,1.1-0.9,2-2,2S6,7.1,6,6V4z" />{" "}
+      <path
+        data-color="color-2"
```
Comment (sampotts) [inline src/components/TextInput/icons.tsx:35]: We should remove these attributes and the className on the `<g>` as they're not needed.

File: src/components/TextInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
+// @ts-nocheck
```
Comment (sampotts) [inline src/components/TextInput/index.tsx:1]: Doesn't this disable TypeScript completely?

File: src/components/TextInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+// tslint:disable: jsx-no-lambda
```
Comment (sampotts) [inline src/components/TextInput/index.tsx:2]: Perhaps we should move to ESLint here?

File: src/assets/icons/empty-mic.svg [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16">
+	<g>
+		<path d="M8,10c2.2,0,4-1.8,4-4V4c0-2.2-1.8-4-4-4S4,1.8,4,4v2C4,8.2,5.8,10,8,10z M6,4c0-1.1,0.9-2,2-2s2,0.9,2,2v2 c0,1.1-0.9,2-2,2S6,7.1,6,6V4z" />
+		{" "}
```
Comment (sampotts) [inline src/assets/icons/empty-mic.svg:4]: I think we can get rid of this? 

Also, is prettier setup for this repo? If not, we should probably set it up for 2 space tabs, etc.

File: src/assets/icons/index.tsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { ReactComponent as MicIcon } from "./mic.svg";
```
Comment (sampotts) [inline src/assets/icons/index.tsx:1]: A small detail but I think we should probably have all the imports at the top and then all the exports?

File: src/assets/icons/mic.svg [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16">
+	<g>
+		<path d="M8,10c2.2,0,4-1.8,4-4V4c0-2.2-1.8-4-4-4C5.8,0,4,1.8,4,4v2C4,8.2,5.8,10,8,10z" />
+		{" "}
```
Comment (sampotts) [inline src/assets/icons/mic.svg:4]: As above

File: src/components/TextInput/index.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+            )}
+            {loading && <SpinnerIcon />}
+          </StyledIconContainer>
+        ) : null}
+      </Box>
+    );
+  }
+);
+
+if(__DEV__) {
```
Comment (sampotts) [inline src/components/TextInput/index.tsx:105]: I think we need to add prettier and perhaps ESLint configs in here to fix this sort of thing - it should be `if (...)` (the space).

**Comment by sampotts** [comment]
@tuanddd Sorry, could you resolve the merge conflicts on this one please? I've corrected the folder name to `assets` (added the s) too.

**Review by sampotts — APPROVED** [review]

**Pull request merged**